notebooks/06a Modelling - classification.ipynb

```diff
@@ -0,0 +1,865 @@
+    "            prop = (subset.pred == 2).sum() / subset.shape[0]\n",
+    "        elif proportion == \"3 - Normal Risk\":\n",
+    "            prop = (subset.pred == 3).sum() / subset.shape[0]\n",
+    "        elif proportion == \"4 - Elevated Risk\":\n",
+    "            prop = (subset.pred == 4).sum() / subset.shape[0]\n",
+    "        else:\n",
+    "            prop = (subset.pred == 5).sum() / subset.shape[0]\n",
+    "\n",
+    "        risks[proportion][this_risk - 1] = prop\n",
+    "\n",
+    "fig = plt.figure()\n",
+    "ax = fig.add_subplot(1, 1, 1)\n",
+    "bottom = np.array([0.0, 0.0, 0.0, 0.0, 0.0])\n",
+    "for proportion in risks:\n",
+    "    if proportion == \"1 - Very Low Risk\":\n",
+    "        data = risks[proportion]\n",
+    "        ax.bar(risk_labels, data, label=proportion, width=0.35)\n",
+    "    else:\n",
+    "        bottom += data\n",
+    "        data = risks[proportion]\n",
+    "        ax.bar(risk_labels, data, label=proportion, bottom=bottom, width=0.35)\n",
+    "handles, labels = ax.get_legend_handles_labels()\n",
+    "ax.legend(handles[::-1], labels[::-1], bbox_to_anchor=(1.05, 1));"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Severity of misclassification\n",
+    "\n",
+    "When the model incorrectly predicts a class, how badly does it do this?\n",
+    "\n",
+    "Because risk categories are numerical (1-5), we can calculate the difference between them as the number of classes incorrect the prediction was."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "gather": {
+     "logged": 1653053346602
+    }
+   },
+   "outputs": [],
+   "source": [
+    "predictions_df[\"diff\"] = predictions_df.diff().risk\n",
+    "fig = px.histogram(predictions_df, x=\"diff\")\n",
+    "fig.show()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "nteract": {
+     "transient": {
+      "deleting": false
+     }
+    }
+   },
+   "source": [
+    "### Feature importance\n",
+    "\n",
+    "Which features does the model ascribe predictive power to?"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "jupyter": {
+     "outputs_hidden": false,
+     "source_hidden": false
+    },
+    "nteract": {
+     "transient": {
+      "deleting": false
+     }
+    }
+   },
+   "outputs": [],
+   "source": [
+    "# Feature names\n",
+    "coef = pd.DataFrame(data=list(X_train.columns))\n",
+    "# Feature importances, sorted\n",
+    "coef[\"coef\"] = models[model][\"model\"].feature_importances_\n",
+    "coef.sort_values(\"coef\", ascending=False, inplace=True)\n",
+    "coef.set_index(0, inplace=True)\n",
+    "# Plot interactive plot\n",
+    "# Hover over a feature for full feature name\n",
+    "fig = px.bar(coef, x=coef.index, y=\"coef\")\n",
+    "fig.show()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
```

Comment:
Will there be binary classification added? Feels like could be potential for that to perform better

Reply:
Great suggestion, could do:

Binary classification Long Stayer (21+ days) or not
->
Regression for Short Stay
OR
Regression for Long Stay

Could we do this for IS_MAJOR?